lib/remap-functions/src/parse_key_value.rs

```diff
@@ -0,0 +1,370 @@
+use nom::{
+    self,
+    branch::alt,
+    bytes::complete::{escaped, tag, take_until, take_while1},
+    character::complete::{char, satisfy, space0},
+    combinator::{map, rest},
+    multi::{many1, separated_list1},
+    sequence::{delimited, preceded, tuple},
+    IResult,
+};
+use remap::prelude::*;
+use std::collections::BTreeMap;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseKeyValue;
+
+impl Function for ParseKeyValue {
+    fn identifier(&self) -> &'static str {
+        "parse_key_value"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "key_value_delimiter",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: false,
+            },
+            Parameter {
+                keyword: "field_delimiter",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+        let key_value_delimiter = arguments.optional("key_value_delimiter").map(Expr::boxed);
+        let field_delimiter = arguments.optional("field_delimiter").map(Expr::boxed);
+
+        Ok(Box::new(ParseKeyValueFn {
+            value,
+            key_value_delimiter,
+            field_delimiter,
+        }))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub(crate) struct ParseKeyValueFn {
+    value: Box<dyn Expression>,
+    key_value_delimiter: Option<Box<dyn Expression>>,
+    field_delimiter: Option<Box<dyn Expression>>,
+}
+
+impl Expression for ParseKeyValueFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let bytes = self.value.execute(state, object)?.try_bytes()?;
+        let value = String::from_utf8_lossy(&bytes);
+
+        let key_value_delimiter = match &self.key_value_delimiter {
+            Some(s) => String::from_utf8_lossy(&s.execute(state, object)?.try_bytes()?).to_string(),
+            None => "=".to_string(),
+        };
+
+        let field_delimiter = match &self.field_delimiter {
+            Some(s) => String::from_utf8_lossy(&s.execute(state, object)?.try_bytes()?).to_string(),
+            None => " ".to_string(),
+        };
+
+        let (_, values) =
+            parse_line(&value, &key_value_delimiter, &field_delimiter).map_err(|e| e.to_string())?;
+
+        Ok(values.into_iter().collect::<BTreeMap<_, _>>().into())
```

Comment:
nit: `Value` implements `FromIterator<(String, Value)>`, so you should be able to do `Ok(Value::from_iter(values))`